libs/platform/src/lib/components/dynamic-page/dynamic-page-header/header/dynamic-page-header.component.ts

```diff
@@ -0,0 +1,405 @@
+
+    /**
+     * subscription for when toggle header is called
+     */
+    toggleSubscription: Subscription;
+    /**
+     * subscription for when expand header is called
+     */
+    expandSubscription: Subscription;
+    /**
+     * subscription for when collapse header is called
+     */
+    collapseSubscription: Subscription;
+    // tabbedSubscription: Subscription;
+
+    /**
+     * @hidden
+     * expand/collapse aria label based on the current state
+     */
+    expandCollapseAriaLabel: string;
+
+    /**
+     * tracking if pin button is pinned
+     */
+    pinned = false;
+
+    /**
+     * @hidden
+     * pn/unpin aria label based on the current state
+     */
+    pinUnpinAriaLabel: string;
+
+    /**
+     *  tracking collapsible for pinning
+     *
+     */
+    _collapsible = this.collapsible;
+
+    /**
+     * @hidden
+     * tracking the background value
+     */
+    private _background: BACKGROUND_TYPE;
+
+    /**
+     * @hidden
+     * tracks the size for responsive padding
+     */
+    private _size: RESPONSIVE_SIZE;
+
+    /** @hidden */
+    constructor(
+        private _elementRef: ElementRef<HTMLElement>,
+        private _renderer: Renderer2,
+        protected _dynamicPageConfig: DynamicPageConfig,
+        private _dynamicPageService: DynamicPageService
+    ) {
+        if (this.collapsible) {
+            this.toggleSubscription = this._dynamicPageService.$toggle.subscribe((val) => {
+                this.toggleCollapse();
+            });
+            this.expandSubscription = this._dynamicPageService.$expand.subscribe(() => {
+                this.collapseHeader(false);
+            });
+            this.collapseSubscription = this._dynamicPageService.$collapse.subscribe(() => {
+                this.collapseHeader(true);
+            });
+        }
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (this._isCollapsibleCollapsed()) {
+            this._setStyleToHostElement('z-index', 1);
+        }
+        this._calculateExpandCollapseAriaLabel();
+        this._calculatePinUnpinAriaLabel();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this._background) {
+            this._setBackgroundStyles(this._background);
+        }
+        if (this.size) {
+            this._setSize(this.size);
+        }
+    }
+
+    /**@hidden */
+    ngOnDestroy(): void {
+        this.toggleSubscription.unsubscribe();
+        this.expandSubscription.unsubscribe();
+        this.collapseSubscription.unsubscribe();
+    }
+
+    collapseHeader(val: boolean): any {
+        if (this._isPinned()) {
+            return;
+        }
+        if (this.collapsed !== val) {
+            this.collapsed = val;
+            this.expandCollapseActions();
+        }
+    }
+
+    /** Handles expanded/collapsed event */
+    toggleCollapse(): void {
+        if (this._isPinned()) {
+            return;
+        }
+        this.collapsed = !this.collapsed;
```

Comment:
What is the source of truth for the `collapsed` state?
Since we have `DynamicPageService` I suppose it should be there.
In the DynamicPageContent you are listening to the `collapsed` changes and try to keep `toggledVal` up to date.
If so we propabbly need to make this kind of changes only through `DynamicPageService`.
Right now if  a user initiates the PageHeader with `collapsed = true` (false by default) the `DynamicPageContent` wll not know about it.

Reply:
maintaining the collapsed state in the service now